src/controllers/ChatsController.ts

```diff
@@ -28,11 +29,38 @@ export class ChatsController {
     }
   }
 
+  checkAndSetParameterInRoute() {
+    const parameter = router.getParameter();
+    if (parameter === null) {
+      store.set('currentChatId', parameter);
+      return parameter;
+    }
+    const validatedId = this.validateChatId(Number(parameter));
+    store.set('currentChatId', validatedId);
+    return validatedId;
+  }
+
+  toGoToChatById(id: number | string | null) {
+    if (id === null) {
+      store.set('currentChatId', null);
+      router.go(Routes.Messenger, null);
+    } else {
+      store.set('currentChatId', id);
+      router.go(Routes.Messenger, Number(id));
+    }
+  }
+
+  validateChatId(id: number) {
+    const { chats } = store.getState() as StateI;
+    const isThereSuchAnIdAmongChats = chats.some(({ id: chatId }) => chatId === id);
+    return isThereSuchAnIdAmongChats ? id : null;
+  }
```

Comment:
Этот метод можно упростить
```suggestion
  chatIdValid(id: number): boolean {
    const { chats } = store.getState() as StateI;
    return chats.some(({ id: chatId }) => chatId === id)
  }
```

Reply:
Тогда опять таки придется делать проверку в checkAndSetParameterInRoute, так как после валидации нам надо установить в стор currentChatId либо id чата если он прошел валидацию, либо null, связанность этих методов и так большая, поэтому я решил сделать подобие пайплайна: приводим к number -> валидируем -> результат записываем в стор.